admin/admin-service/src/main/java/io/mosip/admin/service/impl/ApplicantDetailServiceImpl.java

```diff
@@ -0,0 +1,144 @@
+    private RidValidator<String> ridValidator;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AuditUtil auditUtil;
+
+    @Autowired
+    CbeffToBiometricUtil cbeffToBiometricUtil;
+
+
+    @Autowired
+    Utility utility;
+
+    private static final String IDENTITY = "identity";
+
+    private static final String RESPONSE = "response";
+
+    private static final String DOCUMENTS="documents";
+
+    private static final String ApplicantPhoto = "applicantPhoto";
+
+    private static final String VALUE = "value";
+    private static final String DOB = "dob";
+
+    @Value("${mosip.admin.applicantDetails}")
+    private String[] applicantDetails;
+
+
+    @Override
+    public ApplicantDetailsDto getApplicantDetails(String rid) throws Exception {
+        ApplicantDetailsDto applicantDetailsDto =new ApplicantDetailsDto();
+        Map<String,String> applicantDataMap=new HashMap<>();
+        ConvertRequestDto convertRequestDto = new ConvertRequestDto();
+        List<String> pathsegments=new ArrayList<>();
+        String dateOfBirth=null;
+        String individualBiometrics=null;
+        pathsegments.add(rid);
+        String imageData=null;
+        try {
+			if (!ridValidator.validateId(rid)) {
+				throw new RequestException(ApplicantDetailErrorCode.RID_INVALID.getErrorCode(),
+                        ApplicantDetailErrorCode.RID_INVALID.getErrorMessage());
+			}
+            String response = restClient.getApi(ApiName.RETRIEVE_IDENTITY_API,pathsegments,"type","bio",String.class);
+            JSONObject responseObj= objectMapper.readValue(response,JSONObject.class);
+            if(response!=null && responseObj.containsKey("response")) {
+                JSONObject responseJsonObj=  utility.getJSONObject(responseObj,RESPONSE);
+                JSONObject identityObj=utility.getJSONObject(responseJsonObj,IDENTITY);
+                JSONArray documents=utility.getJSONArray(responseJsonObj,DOCUMENTS);
+                String idenitityJson=utility.getMappingJson();
+                JSONObject idenitityJsonObject=objectMapper.readValue(idenitityJson,JSONObject.class);
+                JSONObject mapperIdentity=utility.getJSONObject(idenitityJsonObject,IDENTITY);
+                List<String> mapperJsonKeys = new ArrayList<>(mapperIdentity.keySet());
+                for (String key: mapperJsonKeys){
+                    for(String valueObj: applicantDetails){
+                        if(valueObj.equalsIgnoreCase(key)){
+                            LinkedHashMap<String, String> jsonObject = utility.getJSONValue(mapperIdentity, key);
+                            String value = jsonObject.get(VALUE);
+                            applicantDataMap.put(value,identityObj.get(value).toString());
+                        }
+
+                    }
+                }
+                for (String applicantDetail: applicantDetails) {
+                    if (applicantDetail.equalsIgnoreCase(ApplicantPhoto)) {
```

Comment:
Can we avoid this for loop and and move this if condition inside the above applicantDetails for loop to avoid one more iteration of the same array ?

Reply:
Also if possible move the code inside this if condition to a different method called, getImageData to reduce the size of this method.

Reply:
taken care

Reply:
Resolved